### PR TITLE
Add user & group functionality to projects

### PR DESCRIPTION
**Pull request by therealgambo**

### CODE REVIEW

File: gitlabform/gitlabform/core.py [hunk 1]
```diff
@@ -256,6 +257,37 @@ def process_merge_requests(self, project_and_group, configuration):
...
             self.gl.put_approvers(project_and_group, approvers, approver_groups)
 
+    @if_in_config_and_not_skipped
+    @configuration_to_safe_dict
+    def process_members(self, project_and_group, configuration):
+        groups = configuration.get('members|groups')
+        if groups:
+            for group in groups:
+                logging.debug("Setting group '%s' as a member", group)
+                access = groups[group]['group_access'] if \
```
Comment (gdubicki) [inline gitlabform/gitlabform/core.py]: `groups[group]['group_access']` has to be defined, according to https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project . I would not introduce own arbitrary default.

Comment (therealgambo) [reply gitlabform/gitlabform/core.py]: This functionality uses a different api endpoint (https://docs.gitlab.com/ee/api/projects.html#share-project-with-group) from the users, however I have grouped them somewhat logically in the configuration.

Comment (therealgambo) [reply gitlabform/gitlabform/core.py]: Any chance of resolving this and merging?

Comment (gdubicki) [reply gitlabform/gitlabform/core.py]: > Any chance of resolving this and merging?

Please see my comment above (I wrote it a while ago already but forgot to click _post_ , sorry).

File: gitlabform/gitlabform/core.py [hunk 2]
```diff
@@ -256,6 +257,37 @@ def process_merge_requests(self, project_and_group, configuration):
...
+                    self.gl.unshare_with_group(project_and_group, group)
+                except NotFoundException:
+                    pass
+                self.gl.share_with_group(project_and_group, group, access, expiry)
+
+        users = configuration.get('members|users')
+        if users:
+            for user in users:
+                logging.debug("Setting user '%s' as a member", user)
+                access = users[user]['access_level'] if \
```
[the remark this answers is not in the document]
Comment (therealgambo) [reply gitlabform/gitlabform/core.py]: Only sending the parameter if it is defined now

File: gitlabform/gitlabform/core.py (hunk 1 above)
Comment (therealgambo) [reply gitlabform/gitlabform/core.py]: Only sending the parameter if it is defined now

**Pull request merged**